### ISSUE TITLE
`META not defined` on unsupported objects in docs.jl

### ISSUE THREAD

**Issue by malmaud** [JuliaLang/julia#11334]
```julia
>> ?GenSym(1)
```

```
LoadError: UndefVarError: META not defined
while loading In[546], in expression starting on line 371

 in doc at docs.jl:28
```

**Comment by vtjnash** [comment]
it seems that `docs.jl` doesn't handle the case very gracefully where the module has no META node defined

although, since GenSym lives in Core, it would need to be documented in the doc folder instead of in the source code. even though GenSym is an internal compiler feature, it is probably still worth documenting since it's pretty visible to any user looking at their code_lowered or code_typed results (possible in the context of a readme on how to interpret these functions?)

**Comment by JeffBezanson** [comment]
It looks like this is partly due to `doc(obj, meta)` being defined for adding doc info, but some other 2-argument `doc` definitions only look up info. I suspect those operations should have different names.
@one-more-minute

**Comment by one-more-minute** [comment]
Fixed (ish) with b307f88cfc8f37f8ec25565a2f2a69dacc9e48bf.

Yes, a function's behaviour shouldn't depend on how it's called. I changed the setter version of `doc` to `doc!` so that it's no longer a heisenfunction. That makes the error more obvious, I suppose.

The main issue here is that `@doc GenSym(1)` tries to get docs for the method `GenSym(::Int)` as opposed to the created object. That's fine in itself, except that the doc system currently doesn't support methods for anything other than functions (see #9260). For now I've added a fallback which simply ignores the method call and returns docs for the `GenSym` type.
